api/api.go

```diff
@@ -0,0 +1,150 @@
+package api
+
+import (
+	"doccer/model"
+	"encoding/json"
+	"github.com/gorilla/mux"
+	"net/http"
+)
+
+type Api struct {
+	useCases model.UseCasesInterface
+}
+func NewApi(x model.UseCasesInterface) *Api {
+	return &Api{
+		useCases: x,
+	}
+}
+
+func (a *Api) Router() http.Handler {
+	router := mux.newRouter
+
+	router.HandleFunc("/register", a.register).Methods(http.MethodPost)
+	router.HandleFunc("/login", a.login).Methods(http.MethodPost)
+	router.HandleFunc("/logout", a.logout).Methods(http.MethodPost)
+
+	router.HandleFunc("/docs/{doc_id}/get", a.getDoc).Methods(http.MethodGet)
```

Comment:
Please use the RESTful approach here, mapping your routes to HTTP verbs. No need to use "/delete" and similar suffixes.